### ISSUE TITLE
TestRunner Log view no longer visible in report

### ISSUE THREAD

**Issue by testphreak** [anshooarora/extentreports-java#716]
Strangely, I no longer see the TestRunner Log view in the report (the one with the clipboard icon). Any ideas what changed? Using v2.41.2. No luck with v2.41.1 either.

My extent-config.xml is pretty basic -
```javascript
<!-- custom javascript -->
        <scripts>
            <![CDATA[
                $(document).ready(function() {
                    $('.dashboard-view').click();
                });
            ]]>
        </scripts>
```

And I make sure to use `setTestRunnerOutput()` like this -
```java
    /**
     * Output test execution logs to ExtentReports
     */
    @AfterSuite(alwaysRun = true)
    public void outputTestExecutionLogs() {
        //log output of tests to TestRunner logs view of ExtentReports html file
        Set<String> deDuplicatedReporterOutput = new LinkedHashSet<>(Reporter.getOutput());
        for (String s : deDuplicatedReporterOutput) {
            ExtentReporter.getExtent().setTestRunnerOutput(s);
        }
    }
```

Any thoughts on what might be the problem?

**Comment by email2vimalraj** [comment]
Did you try debugging whether you had items in your `Reporter.getOutput()`?

Just out of curious, any reason for creating another copy of `LinkedHashSet` of your `Reporter.getOutput()`? You could simply do this:

```java
ExtentReporter.getExtent().setTestRunnerOutput(Reporter.getOutput());
```

Regards
Vimalraj
[http://vimalselvam.com](http://vimalselvam.com)

**Comment by testphreak** [comment]
Yes, I did. `Reporter.getOutput()` was a non-zero number. The reason for using a LinkedHashSet was to remove duplicates as a result of this [issue](https://github.com/cbeust/testng/issues/171), which seems to have been fixed and closed. But using the LinkedHashSet shouldn't have anything do with why the TestRunner log view is not appearing in the report.

**Comment by testphreak** [comment]
Hi @anshooarora I updated to the latest community version (v3.0.3) but I still don't see the option to view TestRunner logs. Any idea what I might be missing. Code snippet above.

**Comment by testphreak** [comment]
Hi @anshooarora can I get an update on this? Am I missing something obvious?

**Comment by anshooarora** [comment]
@testphreak - I tested with 3 simple loc and I do see this view:

```
extent.setTestRunnerOutput("log 1");
extent.setTestRunnerOutput("<pre>Log 2</pre>");
extent.setTestRunnerOutput("<h2>heading 2</h2>");
```

Similar to the [sample report](http://extentreports.com/samples/extent.html).

@email2vimalraj - are you also not seeing the test-runner logs view?

**Comment by email2vimalraj** [comment]
I'm able to see the test runner output view.

**Comment by testphreak** [comment]
Thanks @anshooarora and @email2vimalraj. The problem was related to some changes I had introduced late last year.

For anyone who might run into this issue. The reason why the test-runner logs view wasn't showing up for me was because -
1. I switched from TestNG Reporter to Log4j so `Reporter.getOutput()` was always 0. 
2. I was calling `Extent.getExtent().flush()` in the `@AfterMethod` and I was calling `Extent.getExtent().setTestRunnerOutput()` in the `@AfterSuite`. 

Once I moved the flush operation to the `@AfterSuite` I saw the test-runner view on the report.
